kani-compiler/src/kani_middle/transform/check_uninit/ty_layout.rs

```diff
@@ -0,0 +1,275 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//
+//! Utility functions that help calculate type layout.
+
+use stable_mir::abi::{FieldsShape, Scalar, TagEncoding, ValueAbi, VariantsShape};
+use stable_mir::target::{MachineInfo, MachineSize};
+use stable_mir::ty::{AdtKind, IndexedVal, RigidTy, Ty, TyKind, UintTy};
+use stable_mir::CrateDef;
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+struct DataBytes {
+    /// Offset in bytes.
+    offset: usize,
+    /// Size of this data chunk.
+    size: MachineSize,
+}
+
+pub struct TypeLayout {
```

Comment:
Please document this structure. I'm assuming you are representing the non-padding bytes of a type.

Reply:
BTW, do you even need this structure to be public? It seems that only `to_byte_mask()` is relevant, so why not expose the byte mask instead?

Reply:
> BTW, do you even need this structure to be public? It seems that only `to_byte_mask()` is relevant, so why not expose the byte mask instead?

Good idea, did that ^